### DISCUSSION TITLE
Can I Combine MCMC Posteriors from Multiple Light Curves?

### DISCUSSION THREAD

**Discussion by mzb123** [phoebe-project/phoebe2#834] [Questions]
Hello! I am currently attempting to solve multiple time-series light curves of the same target star. MCMC can provide the distribution and uncertainties of system parameters for each light curve.  However, due to inconsistent upper and lower bounds of the uncertainties, I am facing difficulties in obtaining average system parameters and errors. I have tried to merge the MCMC posterior distributions of system parameters for each light curve (such as orbital inclination, temperature, etc., assuming that system parameters remain constant over a period of time).    However, it cannot be changed due to the samples=[ 1.02501912  0.33571303 70.11061801 ...     23.54326641 53.57718844 10.88150304],  I did not succeed.      Is there a way to achieve this?   Of course, my method may be wrong.

**Comment by kecnry** [comment]
Can you explain your situation a little more - did you run separate MCMC runs for each light curve independently and now want to combine them to get a single set of posteriors?  Are the parameters the same between each run?

**Reply by mzb123** [reply]
I apologize for not explaining clearly. Yes, I independently ran MCMC for each light curve, and the parameters set for each light curve were consistent (including system parameters, as well as niters and nwalkers). Due to the minimal differences in the results from each light curve run, I would like to combine them to obtain a single set of posteriors.

**Reply by kecnry** [reply]
It is possible to combine the distributions after the fact with either "and" or "or" logic, but this does drop the correlations between parameters and assumes equal weighting/contributions between the two light curves. I would have expected this to work directly from phoebe with something like `b.get_distribution_collection(solution=('solution_lc1', 'solution_lc2'), combine='and')`, but that doesn't quite seem to work (maybe we can add support for that eventually).

But you can do it manually, per-parameter, with something like:

```
dist_lc1, = b.get_distribution_collection(solution='solution_lc1')
dist_lc2, = b.get_distribution_collection(solution='solution_lc2')

dist_combined_one_param = dist_lc1.to_univariate('teff@primary') & dist_lc2.to_univariate('teff@primary')
dist_combined_one_param.plot(show=True)
```

Ultimately, though, if you want a full set of posteriors that accounts for all of your data, I would suggest running MCMC over both datasets enabled simultaneously.  You can always start the next sampling run from the final distribution from your existing run(s) to hopefully save some burnin time.
